src/components/TextEditor/TextEditor.js

```diff
@@ -1,76 +1,66 @@
-import React, {Fragment} from 'react';
+import React, {Fragment, useCallback, useState, useMemo} from 'react';
+import cn from 'classnames';
 
 import styles from './TextEditor.less';
 
-export default class TextEditor extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isEditing: false,
-            text: props.text
-        }
-    }
+export default function TextEditor(props) {
+ 
+    const {text, onChangeText, asterisk, placeholder = 'Please enter...'} = props;
 
-    handleChangeText = (event) => {
-        this.setState({
-            text: event.target.value
-        });
-    }
+    const [isEditing, setEditing] = useState(false);
+    const [textEdit, setText] = useState(text);
 
-    handleChangeButton = () => {
-        this.setState({
-            isEditing: true
-        });
-    }
+    const handleChangeText = useCallback(({target}) => {
+        setText(target.value);
+    }, [setText]);
 
-    handleCancelButton = () => {
-        this.setState({
-            isEditing: false,
-            text: this.props.text
-        });
-    }
+    const handleChangeButton = useCallback(() => {
+        setEditing(true);
+    }, [setEditing]);
+    
+    const handleCancelButton = useCallback(() => {
+        setText(text);        
+        setEditing(false);
+    }, [setText, text, setEditing]);
 
-    handleSaveButton = () => {
-        const {onChangeText} = this.props;
-        const {text} = this.state;
+    const handleSaveButton = useCallback(() => {
+        onChangeText(textEdit);
 
-        onChangeText(text);
+        setEditing(false);
+    }, [onChangeText, setEditing, textEdit]);
+        
+    const showText = useMemo(() => {
+        if (!text) {
+            return placeholder;
+        }
+        else {
+            return asterisk ? text.replace(/./g, "*") : text;         
+        }
+    }, [text, placeholder, asterisk]);
 
-        this.setState({
-            isEditing: false
-        });
-    }
+    const textStyle = cn([(!text) && styles.textPlaceholder]);
```

Comment:
no need for wrapping `!text` into brackets and I guess you don't need use `cn` here

Reply:
and it's not `textStyle` it's className